### PR TITLE
[Minor] Improve assert throws tests

### PR DESCRIPTION
**Pull request by justinmclean**
### What changes were proposed in this pull request?

Where possible make sure assertThrows only includes one call that can throw an exception.

### Why are the changes needed?

When test code raises a runtime exception, it is good practice to avoid having multiple method calls inside the assert throws as It can be confusing what is being tested. This change improves the clarity of the tests and avoids potential incorrect testing when another method could raise an exception.

Fix: # N/A

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

Tested locally.

### CODE REVIEW

File: api/src/test/java/com/datastrato/gravitino/TestNameIdentifier.java [hunk 1]
```diff
@@ -41,9 +41,10 @@ public void testCreateWithInvalidArgs() {
     assertThrows(IllegalArgumentException.class, () -> NameIdentifier.of("a", null));
     assertThrows(IllegalArgumentException.class, () -> NameIdentifier.of("a", ""));
 
+    Namespace empty = Namespace.empty();
     assertThrows(IllegalArgumentException.class, () -> NameIdentifier.of(null, "a"));
-    assertThrows(IllegalArgumentException.class, () -> NameIdentifier.of(Namespace.empty(), null));
-    assertThrows(IllegalArgumentException.class, () -> NameIdentifier.of(Namespace.empty(), ""));
+    assertThrows(IllegalArgumentException.class, () -> NameIdentifier.of(empty, null));
```
Comment (mchades) [inline api/src/test/java/com/datastrato/gravitino/TestNameIdentifier.java:46]: Should we verify the error message to confirm that it is the expected error?

Comment (justinmclean) [reply api/src/test/java/com/datastrato/gravitino/TestNameIdentifier.java:46]: I'm not 100% sure what you mean. Checking error strings in tests can be fragile if you are suggesting that.

Comment (mchades) [reply api/src/test/java/com/datastrato/gravitino/TestNameIdentifier.java:46]: For example:
```java
Exception exception =
        assertThrows(
            IllegalArgumentException.class, () -> NameIdentifier.of(Namespace.empty(), null));
    assertTrue(
        exception.getMessage().contains("Cannot create a NameIdentifier with null or empty name"),
        "Message: " + exception.getMessage());
```

Comment (justinmclean) [reply api/src/test/java/com/datastrato/gravitino/TestNameIdentifier.java:46]: This PR was just a refactor to fix how assertThrows was misused; if you think we should check for message text, I suggest you raise another issue for that.

Comment (mchades) [reply api/src/test/java/com/datastrato/gravitino/TestNameIdentifier.java:46]: OK, current changes LGTM.

**Review by yuqi1129 — APPROVED** [review]
It's okay for me.

**Pull request merged**